packages/dataviews/src/view-list.tsx

```diff
@@ -28,24 +37,42 @@ import type {
 	ViewList as ViewListType,
 } from './types';
 
+import { ActionsDropdownMenuGroup } from './item-actions';
+
+interface Action {
+	callback: ( items: Item[] ) => void;
+	hideModalHeader: any;
+	icon: any;
+	id: string;
+	isDestructive: boolean | undefined;
+	isEligible: any;
+	isPrimary: boolean;
+	label: string;
+	modalHeader: string;
+	RenderModal: any;
+}
+
 interface ListViewProps {
-	view: ViewListType;
-	fields: NormalizedField[];
+	actions: Action[];
 	data: Data;
-	isLoading: boolean;
+	fields: NormalizedField[];
 	getItemId: ( item: Item ) => string;
+	id: string;
+	isLoading: boolean;
 	onSelectionChange: ( selection: Item[] ) => void;
 	selection: Item[];
-	id: string;
+	view: ViewListType;
 }
 
 interface ListViewItemProps {
+	actions: Action[];
 	id?: string;
-	item: Item;
 	isSelected: boolean;
-	onSelect: ( item: Item ) => void;
+	item: Item;
 	mediaField?: NormalizedField;
+	onSelect: ( item: Item ) => void;
 	primaryField?: NormalizedField;
+	store: any;
```

Comment:
Unsure if there's a better way to type this https://ariakit.org/reference/use-composite-store

Reply:
`import type { CompositeStore } from '@ariakit/react';`  should do it. We actually export this type from the v2 composite module (and you can see it used [here](https://github.com/WordPress/gutenberg/blob/1d6e92b2403a24563562ef85676b6d01ebe6098b/packages/components/src/circular-option-picker/types.ts#L126)), but I don't think we re-export it from the components package. @mirka any idea if this is intentional? I'd be fine to export the type.

Reply:
I've been [trying to avoid](https://github.com/WordPress/gutenberg/issues/60193#issuecomment-2024740611) us having to export types from the package, to reduce maintenance cost. The suggested way is to extract the types from the components/hooks that we export.

```ts
type CompositeStore = React.ComponentProps< typeof Composite >[ 'store' ];
type CompositeStoreProps = Parameters< typeof useCompositeStore >[ 0 ];
```

Though, I was looking at this specific case and noticed that the `unlock()` function for private APIs is typed in a way that doesn't preserve the original types of the locked objects. I'll look into that.